tests/test_modeling_common.py

```diff
@@ -595,6 +595,47 @@ def test_inputs_embeds(self):
             with torch.no_grad():
                 model(**inputs_dict)
 
+    def test_lm_head_model_random_generate(self):
+
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        input_ids = inputs_dict.get(
+            "input_ids", None
+        )  # TODO (PVP): ugly workaround to make code work for t5 for the moment - has to changed when t5 is fixed.
+
+        for model_class in self.all_generative_model_classes:
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+
+            if config.bos_token_id is None:
+                with self.assertRaises(AssertionError):
+                    model.generate(max_length=5)
```

Comment:
decided to "hard-code" the variables `max_length, num_beams and num_return_sequences` here. The reason is that the values are not specific to the model at all and (in my opinion) won't have to change if a new lm model is added (a new model has to be able to do beam_generation & has to be able to generate > 1 sequences and also has to work for max_length=5). More specific cases that resemble the Integration tests can then be tested using a separate function in the respective model files. @LysandreJik

Reply:
I agree with that assumption.